Build error: PROTOBUF_ARRAY_SIZE

**Describe the bug**
I'm getting a consistent build on current master when I try to build basically anything:

```
compiling float64.pb.cc failed.
In file included from /Users/kevinkillingsworth/projects/robotics/allwpilib/simulation/gz_msgs/build/generated/main/simulation/gz_msgs/float64.pb.cc:4:
In file included from /Users/kevinkillingsworth/projects/robotics/allwpilib/simulation/gz_msgs/build/generated/main/simulation/gz_msgs/float64.pb.h:26:
In file included from /usr/local/include/google/protobuf/generated_message_table_driven.h:35:
In file included from /usr/local/include/google/protobuf/map_entry_lite.h:42:
/usr/local/include/google/protobuf/generated_message_util.h:230:14: error: zero size arrays are an extension [-Werror,-Wzero-length-array]
  void* deps[PROTOBUF_ARRAY_SIZE(N)];
             ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/google/protobuf/generated_message_util.h:215:32: note: expanded from macro 'PROTOBUF_ARRAY_SIZE'
#define PROTOBUF_ARRAY_SIZE(n) (n)
                               ^~~
/Users/kevinkillingsworth/projects/robotics/allwpilib/simulation/gz_msgs/build/generated/main/simulation/gz_msgs/float64.pb.cc:36:47: note: in instantiation of template class 'google::protobuf::internal::SCCInfo<0>' requested here
::PROTOBUF_NAMESPACE_ID::internal::SCCInfo<0> scc_info_Float64_float64_2eproto =
                                              ^
1 error generated.

compiling bool.pb.h successful.
```

**To Reproduce**
Steps to reproduce the behavior:
Using OSX Catalina 10.15.4
1. `./gradlew build` and this happens several times

**Expected behavior**
A successful build

**Screenshots**
See above compiler output

**Desktop (please complete the following information):**
 - OS: Mac OSX Catalina 10.15.4

```
$ java --version
java 13.0.1 2019-10-15
Java(TM) SE Runtime Environment (build 13.0.1+9)
Java HotSpot(TM) 64-Bit Server VM (build 13.0.1+9, mixed mode, sharing)
```

```
$ gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

--- a/simulation/frc_gazebo_plugins/build.gradle
+++ b/simulation/frc_gazebo_plugins/build.gradle
@@ -21,16 +21,14 @@ try {
     gazebo_linker_args = "pkg-config --libs gazebo protobuf".execute().text.split()
 } catch(Exception ex) { }
 
-if (!gazebo_version?.trim()) {
-    println "Gazebo development files are not available. (pkg-config --modversion gazebo failed)"
-    if (project.hasProperty("makeSim")) {
-        /* Force the build even though we did not find protobuf. */
+if (project.hasProperty("makeSim")) {
+    if (!gazebo_version?.trim()) {
+        println "Gazebo development files are not available. (pkg-config --modversion gazebo failed)"
         println "makeSim set. Forcing build - failure likely."
     }
-    else {
-        ext.skip_frc_plugins = true
-        println "Skipping FRC Plugins."
-    }
+} else {
+    ext.skip_frc_plugins = true
+    println "Skipping FRC Plugins."
 }
 
 evaluationDependsOn(":simulation:gz_msgs")

--- a/simulation/gz_msgs/build.gradle
+++ b/simulation/gz_msgs/build.gradle
@@ -26,17 +26,14 @@ try {
 } catch(Exception ex) {
 }
 
-if (!protobuf_version?.trim()) {
-    println "Protobuf is not available. (pkg-config --modversion protobuf failed)"
-    protobuf_version = "+"
-    if (project.hasProperty("makeSim")) {
-        /* Force the build even though we did not find protobuf. */
+if (project.hasProperty("makeSim")) {
+    if (!protobuf_version?.trim()) {
+        println "Protobuf is not available. (pkg-config --modversion protobuf failed)"
         println "makeSim set. Forcing build - failure likely."
     }
-    else {
-        ext.skip_gz_msgs = true
-        println "Skipping gz_msgs."
-    }
+} else {
+    ext.skip_gz_msgs = true
+    println "Skipping gz msgs."
 }
 
 tasks.whenTaskAdded { task ->
@@ -88,6 +85,11 @@ model {
             binaries {
                 all {
                     cppCompiler.args "-fPIC"
+
+                    // Disable -Wzero-length-array on Clang
+                    if (it.targetPlatform.operatingSystem.isMacOsX()) {
+                        it.cppCompiler.args.add('-Wno-error=zero-length-array')
+                    }
                 }
             }
         }

--- a/simulation/halsim_gazebo/build.gradle
+++ b/simulation/halsim_gazebo/build.gradle
@@ -19,16 +19,14 @@ try {
     gazebo_linker_args = "pkg-config --libs gazebo protobuf".execute().text.split()
 } catch(Exception ex) { }
 
-if (!gazebo_version?.trim()) {
-    println "Gazebo development files are not available. (pkg-config --modversion gazebo failed)"
-    if (project.hasProperty("makeSim")) {
-        /* Force the build even though we did not find protobuf. */
+if (project.hasProperty("makeSim")) {
+    if (!gazebo_version?.trim()) {
+        println "Gazebo development files are not available. (pkg-config --modversion gazebo failed)"
         println "makeSim set. Forcing build - failure likely."
     }
-    else {
-        ext.skip_frc_plugins = true
-        println "Skipping FRC Plugins."
-    }
+} else {
+    ext.skip_frc_plugins = true
+    println "Skipping FRC Plugins."
 }
 
 evaluationDependsOn(":simulation:gz_msgs")

